### PR TITLE
Implement startGame()

### PR DESCRIPTION
**Pull request by sarahkatherine**
Implement and Javadoc startGame()

### CODE REVIEW

File: java/src/shared/models/game/ClientModel.java [hunk 1]
```diff
@@ -93,8 +94,33 @@ public ClientModel(@NotNull MessageList chat,
...
+    /**
+     * Constructor that represents a new Client Model for a newly initialized game
+     *
+     * @param randomTiles whether the HexTypes are random
+     * @param randomPorts whether the Port Types are random
+     * @param randomNumbers whether the Numbers on the Hexes are random
+     */
+    public ClientModel(boolean randomTiles, boolean randomPorts, boolean randomNumbers) {
+        setChat(new MessageList());
+
```
Comment (kupiakos) [inline java/src/shared/models/game/ClientModel.java:108]: Could probably remove most of these spaces; not a big deal though.

File: java/src/shared/models/game/GameMap.java [hunk 2]
```diff
@@ -77,6 +80,118 @@ public GameMap(Map<EdgeLocation, PlayerIndex> roads,
...
+     *
+     * @param randomTiles   whether the HexTypes are random
+     * @param randomNumbers whether the Numbers on the Hexes are random
+     */
+    public void setNewGameHexes(boolean randomTiles, boolean randomNumbers) {
+        List<Integer> hexNumbers = setHexNumbers(randomNumbers);
+
+        List<HexType> hexTypes = setHexTypes(randomTiles);
+
+        ArrayList<HexLocation> hexLocations = setHexLocations();
```
Comment (kupiakos) [inline java/src/shared/models/game/GameMap.java:107]: Don't return or assign `ArrayList`, return and assign `List`.

File: java/src/shared/models/game/GameMap.java [hunk 3]
```diff
@@ -77,6 +80,118 @@ public GameMap(Map<EdgeLocation, PlayerIndex> roads,
...
+            hexes.put(hexLocations.get(i), new Hex(hexNumbers.get(i), hexLocations.get(i), hexTypes.get(i)));
+        }
+    }
+
+    /**
+     * Sets the hex locations of the new game based on the radius
+     *
+     * @return the List of Hex Locations for each hex on the map
+     */
+    public ArrayList<HexLocation> setHexLocations() {
```
Comment (kupiakos) [inline java/src/shared/models/game/GameMap.java:119]: Return a `List` not an `ArrayList`

File: java/src/shared/models/game/GameMap.java [hunk 4]
```diff
@@ -77,6 +80,118 @@ public GameMap(Map<EdgeLocation, PlayerIndex> roads,
...
+        setNewGamePorts(randomPorts);
+    }
+
+    /**
+     * Sets the hexes of the new game based on if the user wants random hex types or random numbers on hexes
+     *
+     * @param randomTiles   whether the HexTypes are random
+     * @param randomNumbers whether the Numbers on the Hexes are random
+     */
+    public void setNewGameHexes(boolean randomTiles, boolean randomNumbers) {
```
Comment (kupiakos) [inline java/src/shared/models/game/GameMap.java:102]: Should not be public.

File: java/src/shared/models/game/GameMap.java [hunk 5]
```diff
@@ -77,6 +80,118 @@ public GameMap(Map<EdgeLocation, PlayerIndex> roads,
...
+        }
+        return hexLocations;
+    }
+
+    /**
+     * Sets the Hex Types for each hex on the new game map
+     * @param randomTiles whether the HexTypes are random
+     * @return List of Hex Types that contain a hex type for each hex on the new game map
+     */
+    public List<HexType> setHexTypes(boolean randomTiles) {
```
Comment (kupiakos) [inline java/src/shared/models/game/GameMap.java:135]: Should be private.

File: java/src/shared/models/game/GameMap.java [hunk 6]
```diff
@@ -77,6 +80,118 @@ public GameMap(Map<EdgeLocation, PlayerIndex> roads,
...
+        return hexTypes;
+    }
+
+    /**
+     * Set the Hex Numbers for each hex on the new game map
+     * @param randomNumbers whether the Numbers on the Hexes are random
+     * @return List of Integers that contain the numbers for each hex on the new game map
+     */
+    public List<Integer> setHexNumbers(boolean randomNumbers) {
+        List<Integer> hexNumbers = Arrays.asList(5, 2, 6, 8, 10, 9, 3, -1, 3, 11, 4, 8, 4, 6, 5, 10, 11, 12, 9);
```
Comment (kupiakos) [inline java/src/shared/models/game/GameMap.java:153]: I'm pretty sure desert hexes are supposed to be 0.

File: java/src/shared/models/game/GameMap.java [hunk 7]
```diff
@@ -77,6 +80,118 @@ public GameMap(Map<EdgeLocation, PlayerIndex> roads,
...
+
+        return hexNumbers;
+    }
+
+    /**
+     * Sets the ports of the new game based on if the user wants random port types
+     * @param randomPorts whether the Port Types are random
+     */
+    public void setNewGamePorts(boolean randomPorts) {
+        final List<EdgeLocation> portLocations = Arrays.asList(
```
Comment (kupiakos) [inline java/src/shared/models/game/GameMap.java:167]: Possible support for different radii in the future.

File: java/src/shared/models/game/GameMap.java [hunk 8]
```diff
@@ -77,6 +80,118 @@ public GameMap(Map<EdgeLocation, PlayerIndex> roads,
...
+    /**
+     * Constructor that creates GameMap based on if the user wants random hex types, random port types,
+     * and/or random numbers on hexes
+     *
+     * @param randomTiles   whether the HexTypes are random
+     * @param randomPorts   whether the Port Types are random
+     * @param randomNumbers whether the Numbers on the Hexes are random
+     */
+    public GameMap(boolean randomTiles, boolean randomPorts, boolean randomNumbers) {
+        setNewGameHexes(randomTiles, randomNumbers);
```
Comment (kupiakos) [inline java/src/shared/models/game/GameMap.java:92]: Are we ensuring that the desert hex always gets no number assigned, and that every other hex gets a valid number?

**Review by kupiakos — COMMENTED** [review]
Overall, pretty good. Some changes for correctness and proper style are needed, and the only possible real problem I see is handling numbers with the desert hex.

File: java/src/shared/models/game/GameMap.java (hunk 7 above)
Comment (sarahkatherine) [reply java/src/shared/models/game/GameMap.java:167]: There is no different radii in the future.

File: java/src/shared/models/game/GameMap.java (hunk 5 above)
Comment (sarahkatherine) [reply java/src/shared/models/game/GameMap.java:135]: How would I know that?

Comment (kupiakos) [reply java/src/shared/models/game/GameMap.java:135]: Will it (or should it) ever be used outside of the own class? If the answer is no, then make it private.

**Pull request merged**